packages/api-svc/src/routes.ts

```diff
@@ -399,10 +408,24 @@ export class ExpressRoutes {
 		const batchName = req.query.batchName as string || req.query.batchname as string;
 		const transferId = req.query.transferId as string || req.query.transferid as string;
 		const matrixId = req.query.matrixId as string || req.query.matrixid as string;
+
+		// Optional pagination
+		const pageIndexStr = req.query.pageIndex as string || req.query.pageindex as string;
+		const pageIndex = pageIndexStr ? parseInt(pageIndexStr) : undefined;
+
+		const pageSizeStr = req.query.pageSize as string || req.query.pagesize as string;
+		const pageSize = pageSizeStr ? parseInt(pageSizeStr) : undefined;
+
 		try {
 			let settlementTransfers:ISettlementBatchTransfer[];
 			if (batchId) {
-				settlementTransfers = await this._batchTransferRepo.getBatchTransfersByBatchIds([batchId]);
+				const result = await this._batchTransferRepo.getBatchTransfersByBatchIds([batchId], pageIndex, pageSize);
+				if (!result.items || result.items.length <= 0) {
```

Comment:
Shouldn't this assign to settlementTransfers and be responded at the bottom, like the other if branches?